src/libcore/str/mod.rs

```diff
@@ -1074,46 +1075,44 @@ unsafe fn cmp_slice(a: &str, b: &str, len: usize) -> i32 {
 }
 
 /*
-Section: Misc
+Section: UTF-8 validation
 */
 
+// use truncation to fit u64 into usize
+const NONASCII_MASK: usize = 0x80808080_80808080u64 as usize;
```

Comment:
Didn't think much about it, but the second constant would be in the function by necessity so that it's not defined too far from its use.